session/bootstrap.go

```diff
@@ -1186,7 +1187,9 @@ func syncUpgradeState(s Session) {
 		if i == retryTimes-1 {
```

Comment:
Normally, crash inside a loop is not a good practice, which we may involve more action on the crash later on.

We could do it like :
```
isSuccess = true
for ... {
    ...
    if err == nil ... {
        break
    }

    isSuccess = false
}

if !isSuccess {
    // fatal actions ...   
    ...
    
    logutil.BgLogger().Fatal...
}

```

Reply:
No big deal though.

Reply:
Okay, I won't change it.